src/cargo/util/toml/mod.rs

```diff
@@ -2196,7 +2281,18 @@ impl<P: ResolveToPath + Clone> TomlDependency<P> {
                     label, label
                 )).map(|dep| {
                     match dep {
-                        TomlDependency::Simple(s) => TomlDependency::Simple(s),
+                        TomlDependency::Simple(s) => {
```

Comment:
> Fixed problem where a package would try to pull a dependency from a workspace and specify { workspace = true, optional = true } and it would not respect the optional

Personally, I'd prefer to see unrelated fixes in a separate PR or at least a separate commit to make it easier to review, `git blame`, etc.

Reply:
I will add it as a separate commit on here since one of the tests depends on it.